scripts/veloxbench/veloxbench/_benchmark.py

```diff
@@ -0,0 +1,228 @@
+        # This should work, though it would be much better to get the sha from velox directly so we know we're using the right one.
+        "velox_git_revision": subprocess.check_output(["git", "rev-parse", "HEAD"])
+        .decode("ascii")
+        .strip(),
+    }
+
+
+class Benchmark(conbench.runner.Benchmark):
+    arguments = []
+    options = {"cpu_count": {"type": int}}
+
+    def __init__(self):
+        super().__init__()
+
+    @functools.cached_property
+    def velox_info(self) -> Dict[str, Any]:
+        return velox_info()
+
+    @functools.cached_property
+    def velox_info_r(self) -> Dict[str, Any]:
+        return self._get_velox_info_r()
+
+    @functools.cached_property
+    def github_info(self) -> Dict[str, Any]:
+        return github_info(self.velox_info)
+
+    def benchmark(
+        self,
+        f,
+        extra_tags: Dict[str, Any],
+        options: Dict[str, Any],
+        case: Optional[tuple] = None,
+    ):
+        cpu_count = options.get("cpu_count", None)
+        if cpu_count is not None:
+            # TODO: how to set multithreading?
+            raise NotImplementedError()
+        tags, info, context = self._get_tags_info_context(case, extra_tags)
+        try:
+            benchmark, output = self.conbench.benchmark(
+                f,
+                self.name,
+                tags=tags,
+                info=info,
+                context=context,
+                github=self.github_info,
+                options=options,
+                publish=os.environ.get("DRY_RUN") is None,
+            )
+        except Exception as e:
+            benchmark, output = self._handle_error(e, self.name, tags, info, context)
+        return benchmark, output
+
+    def record(
+        self,
+        result,
+        extra_tags: Dict[str, Any],
+        extra_info,
+        extra_context,
+        options: Dict[str, Any],
+        case: Optional[tuple] = None,
+        name: Optional[str] = None,
+        output=None,
+    ):
+        if name is None:
+            name = self.name
+        tags, info, context = self._get_tags_info_context(case, extra_tags)
+        info.update(**extra_info)
+        context.update(**extra_context)
+
+        benchmark, output = self.conbench.record(
+            result,
+            name,
+            tags=tags,
+            info=info,
+            context=context,
+            github=self.github_info,
+            options=options,
+            output=output,
+            publish=os.environ.get("DRY_RUN") is None,
+        )
+        return benchmark, output
+
+    def execute_command(self, command, capture_output=True):
+        try:
+            print(command)
+            result = subprocess.run(command, capture_output=capture_output, check=True)
+        except subprocess.CalledProcessError as e:
+            print(e.stderr.decode("utf-8"))
+            raise e
+        # Some benchmarks (e.g., java-micro) produce 12GB+ of stdout that can't be loaded into memory
```

Comment:
I don't think our benchmarks produce any output for now

Reply:
_nods_ I can pull this out, it's part of the scaffolding I brought over from the Arrow benchmarks code, but we'll probably not use it here